generated/nidcpower/session.py

```diff
@@ -4151,6 +4151,599 @@ def export_attribute_configuration_file(self, file_path):
+            },
+            "voltage_limit_high": {
+                'id': 1150185,
+                'property': self.voltage_limit_high,
+            },
+            "voltage_limit_low": {
+                'id': 1150186,
+                'property': self.voltage_limit_low,
+            },
+            "current_limit_high": {
+                'id': 1150187,
+                'property': self.current_limit_high,
+            },
+            "current_limit_low": {
+                'id': 1150188,
+                'property': self.current_limit_low,
+            },
+            "pulse_voltage_limit_high": {
+                'id': 1150189,
+                'property': self.pulse_voltage_limit_high,
+            },
+            "pulse_voltage_limit_low": {
+                'id': 1150190,
+                'property': self.pulse_voltage_limit_low,
+            },
+            "pulse_bias_voltage_limit_high": {
+                'id': 1150191,
+                'property': self.pulse_bias_voltage_limit_high,
+            },
+            "pulse_bias_voltage_limit_low": {
+                'id': 1150192,
+                'property': self.pulse_bias_voltage_limit_low,
+            },
+            "pulse_current_limit_high": {
+                'id': 1150193,
+                'property': self.pulse_current_limit_high,
+            },
+            "pulse_current_limit_low": {
+                'id': 1150194,
+                'property': self.pulse_current_limit_low,
+            },
+            "pulse_bias_current_limit_high": {
+                'id': 1150195,
+                'property': self.pulse_bias_current_limit_high,
+            },
+            "pulse_bias_current_limit_low": {
+                'id': 1150196,
+                'property': self.pulse_bias_current_limit_low,
+            },
+            "voltage_level": {
+                'id': 1250001,
+                'property': self.voltage_level,
+            },
+            "ovp_enabled": {
+                'id': 1250002,
+                'property': self.ovp_enabled,
+            },
+            "ovp_limit": {
+                'id': 1250003,
+                'property': self.ovp_limit,
+            },
+            "current_limit": {
+                'id': 1250005,
+                'property': self.current_limit,
+            },
+            "output_enabled": {
+                'id': 1250006,
+                'property': self.output_enabled,
+            },
+        }
+
+        # First we need to get all possible properties we might be setting
+        attribute_ids_used = set()
+        for s in sequence:
+            for key in s:
+                attribute_ids_used.add(attribute_lookup[key]['id'])
+
+        # Create the step with the list of attr ids we have
+        self._create_advanced_sequence(sequence_name, list(attribute_ids_used), set_as_active_sequence)
+
+        for s in sequence:
+            self._create_advanced_sequence_step()
+            for key in s:
```

Comment:
Why not `for key, value in s.items()` to avoid having to lookup the value when we could already have it?

Reply:
Fixed.